dynamoDbQueueSource.js

```diff
@@ -1,76 +1,106 @@
 'use strict';
 
 var _ = require('lodash');
-var AWS = require('aws-sdk');
-
-var dynamoDb = new AWS.DynamoDB();
-
-var dynamoDbQueueSource = { };
-
-dynamoDbQueueSource._processScanResult = function(err, data, scanTask) {
-  // Did we have an error? If so, pass it through to the callback, indicate
-  // that we should stop iterating, and return
-  if (err) {
-    scanTask.error = err;
-    scanTask.callback(err);
-    return;
-  }
-
-  // Push the items from the scan into the queue and update the total item count
-  scanTask.queue.push(data.Items);
-  scanTask.results.itemCount += data.Items.length;
-
-  // Are we done? If so, trigger the callback for the scan and return
-  if (!data.LastEvaluatedKey) {
-    scanTask.hasMoreItems = false;
-    scanTask.callback(null, scanTask.results);
-    return;
-  }
-
-  // If we have more rows to scan, repeat the scan with the new exclusive
-  // start key
-  scanTask.options.ExclusiveStartKey = data.LastEvaluatedKey;
-  setImmediate(_.partial(dynamoDbQueueSource._executeScanTask, scanTask));
-};
 
-dynamoDbQueueSource._executeScanTask = function(scanTask) {
-  dynamoDb.scan(scanTask.options, scanTask.processor);
-};
+module.exports = function(dynamoDb) {
+  var dynamoDbQueueSource = { };
+
+  dynamoDbQueueSource._processResult = function(err, data, task) {
+    // Did we have an error? If so, pass it through to the callback, indicate
+    // that we should stop iterating, and return
+    if (err) {
+      task.error = err;
+      task.callback(err);
+      return;
+    }
+
+    // Push the items from the scan into the queue and update the total item
```

Comment:
The comments are slightly out of date, they only refer to the scan but this function processes results for scans and queues now.